packages/sdk/src/evm/core/classes/contract-sales.ts

```diff
@@ -38,7 +38,11 @@ export class ContractPrimarySale<TContract extends IPrimarySale>
    * @twfeature PrimarySale
    */
   public async getRecipient(): Promise<Address> {
-    return await this.contractWrapper.readContract.primarySaleRecipient();
+    const result = await this.contractWrapper.read<
+      IPrimarySale,
+      "primarySaleRecipient"
+    >("primarySaleRecipient", []);
+    return result;
```

Comment:
oof yeah that's ugly. how often do you think we'll need this workaround? not clear what's causing the type issue